release-tools/keyrotation/key-rotation.py

```diff
@@ -0,0 +1,237 @@
+            print("GIT owner : " + git_owner)
+            repos = source_data.get("key-rotate").get("odfe-repos")
+            for repo in repos:
+                try:
+                    print(repo)
+                    logging.info("Key rotation is in progress for Repo : " + repo.get("repo"))
+                    logging.info("Repo configuration : ")
+                    logging.info(repo)
+                    print("Repo name: " + repo.get("repo"))
+                    git_repo = repo.get("repo")
+                    logging.info("IAM username: " + repo.get("aws_iam_user"))
+                    print("IAM username: " + repo.get("aws_iam_user"))
+                    num_of_keys = number_of_keys(repo.get("aws_iam_user"))
+                    logging.info("Number of secret access keys for " + repo.get("aws_iam_user") + " : " + str(num_of_keys))
+                    print("Number of secret access keys for " + repo.get("aws_iam_user") + " : " + str(num_of_keys))
+                    if num_of_keys == 0:
+                        logging.info("Number of secret access keys are 0")
+                        print("Number of secret access keys are 0")
+                        create_key(repo.get("aws_iam_user"),git_repo,git_owner,repo.get("git_access_key_id"),repo.get("git_access_secret_key"))
+                    elif num_of_keys == 1:
+                        logging.info("Number of secret access keys are 1")
+                        print("Number of secret access keys are 1")
+                        dict = list_keys(repo.get("aws_iam_user"), 1)
+                        logging.info(dict)
+                        deactivate_key(repo.get("aws_iam_user"),dict.get('access_key_id'))
+                        logging.info("Access key has been deactivated : " + dict.get('access_key_id'))
+                        print("Access key has been deactivated : " + dict.get('access_key_id'))
+                        create_key(repo.get("aws_iam_user"),git_repo,git_owner,repo.get("git_access_key_id"),repo.get("git_access_secret_key"))
+                    else:
+                        logging.info("Number of secret access keys are 2")
+                        print("Number of secret access keys are 2")
+                        dict = list_keys(repo.get("aws_iam_user"), 2)
+                        dict_values = list(dict)
+                        logging.info(dict_values)
+                        logging.info("Delete Access key : " + dict_values[0].get('access_key_id'))
+                        print("Delete Access key : " + dict_values[0].get('access_key_id'))
+                        delete_key(repo.get("aws_iam_user"),dict_values[0].get('access_key_id'))
+                        deactivate_key(repo.get("aws_iam_user"),dict_values[1].get('access_key_id'))
+                        create_key(repo.get("aws_iam_user"),git_repo,git_owner,repo.get("git_access_key_id"),repo.get("git_access_secret_key"))
```

Comment:
Create the new key before deactivating the old one.

Reply:
Will make the change and validate.

Reply:
Made the change.